integrations/spotter.h

```diff
@@ -0,0 +1,31 @@
+#include "util.h"
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define USE_TIMESTAMP 1
+#define NO_TIMESTAMP 0
+
+typedef enum {
+  BmNetworkTypeCellularIriFallback = (1 << 0),
+  BmNetworkTypeCellularOnly = (1 << 1),
+} BmSerialNetworkType;
+
+BmErr bm_fprintf(uint64_t target_node_id, const char *file_name,
+                 uint8_t print_time, const char *format, ...);
```

Comment:
Because `bm_fprintf` publishes `spotter/printf` or `spotter/fprintf` under the hood, this one is Spotter-specific, and as you were proposing yesterday @matt001k, I like the idea of changing the function name to make that clear. I think I like `spotter_log` best, which is what I used in the [`bm_serial` CircuitPython implementation](https://github.com/bristlemouth/bm_serial/tree/develop/circuitpython).

Per our discussion though we also don't want to break existing code that calls `bm_fprintf`. So I'd propose that we do the following:
- add a function named `spotter_log` with this code functionality
- keep `bm_fprintf`
- add a doxygen code comment and API docs about `bm_fprintf` being deprecated in favor of `spotter_log`
- have `bm_fprintf` log (with printf or `bm_debug` or whatever, light touch, only see it if you're connected over USB) something about this function being deprecated, please use spotter_log instead, and then call spotter_log.

Thoughts?

Reply:
Another option is to add an attribute `__attribute__((deprecated))` to throw an error when building. Which could then point to the new API to use.
Thoughts on this?

Reply:
Ooooh interesting! I didn't know about that. Yes, 👍 let's do it.

Reply:
![image](https://github.com/user-attachments/assets/613ab761-c3c6-42ec-8dd0-05dcf670c8ef)
This is how the error would read.